enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+            status: False
+          metadata: |
+            {
+              "version": "4.13.7",
+              "arch": "x86_64",
+              "size": "16 GiB",
+              "release": "quay.io/openshift-release-dev/ocp-release@sha256:...",
+              "images": [
+                "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+                "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+                ...
+              ]
+            }
+```
+
+At that point the `bundle-extractor` job will finish.
+
+The `extracted` field for each node will indicate if the bundle extraction
+process has been completed.
+
+The `metadata` field for each node will contain the information from the
+`metadata.json` file of the bundle. Note that the list of images may be too long
+(approximately 180 images) to store it in the status of the `ClusterVersion`
+object; it may be convenient to store them in separate configmaps, and have only
+the references to those configmaps in the `ClusterVersion` status.
+
+When all the nodes have the bundle extracted (the `Extracted` condition for all
+nodes is `True`) the cluster version operator will delete the `bundle-server`
+daemon set and verify that the metadata in all nodes (the content of the
+`metadata` field) is the same. If there are differences they will be reported
+as error conditions in the status of the `ClusterVersion` object and the
+upgrade process will be aborted. This is intended to prevent accidents like
+having two different USB sticks with different bundles plugged in two different
+nodes.
+
+Once the metadata has been validated the cluster version operator will ask the
+machine config operator to pin and pre-load the images specified in the
+metadata. To do so it will use the new `pinnedImages` and `imagesDirectory`
+fields of the `ContainerRuntimeConfig` object:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: ContainerRuntimeConfig
+metadata:
+  name: pin-upgrade
+spec:
+  containerRuntimeConfig:
+    pinnedImages:
+    - quay.io/openshift-release-dev/ocp-release@sha256:...
+    - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+    - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
+    imagesDirectory: /var/lib/upgrade/4.13.7-x86_64
+```
+
+The machine config operator will react to that pinning and pre-loading the
+images in all the nodes of the cluster, as described in the previous section.
+
+When the all the images are pinned and pre-loaded the cluster version operator
+will check the `spec.desiredUpdate.schedule.scheduleType` field of the
+`ClusterVersion` object. This field will be used to indicate if the upgrade has
+to be started manually later or it if should wait till the user explicitly sets
+it to `Immediate`. The default value for that will be `Immediate`. This is
+intended for situations where the user wishes to prepare everything for the
+upgrade in advance, but wants to perform the upgrade later.
```

Comment:
I do not see a reason around why the update schedule needs to be part of this enhancement. We are planning introduce too many changes and some of the changes directly not related to the goal. The update schedule is a generic feature which should be a separate enhancement.